pkg/tkctl/executor/executor.go

```diff
@@ -69,21 +68,50 @@ func (t *PodExecutor) Execute() error {
 	return t.attachPod(pod)
 }
 
+// podRunningAndReady returns true if the pod is running and ready, false if the pod has not
+// yet reached those states, returns ErrPodCompleted if the pod has run to completion, or
+// an error in any other case.
+func podRunningAndReady(event watchapi.Event) (bool, error) {
+	switch event.Type {
+	case watchapi.Deleted:
+		return false, errors.NewNotFound(schema.GroupResource{Resource: "pods"}, "")
+	}
+	switch t := event.Object.(type) {
+	case *v1.Pod:
+		switch t.Status.Phase {
+		case v1.PodFailed, v1.PodSucceeded:
+			return false, cmdrun.ErrPodCompleted
+		case v1.PodRunning:
+			conditions := t.Status.Conditions
+			if conditions == nil {
+				return false, nil
+			}
+			for i := range conditions {
+				if conditions[i].Type == v1.PodReady &&
+					conditions[i].Status == v1.ConditionTrue {
+					return true, nil
+				}
+			}
+		}
+	}
+	return false, nil
+}
```

Comment:
`kubectl.PodRunningAndReady` is not public now, so I had to copy its implementation here